src/frontend/js/classes/sidebar-filter.js

```diff
@@ -0,0 +1,341 @@
+        this.setSectionCollapsed(parentSection, false);
+      }
+    }
+
+    // scroll to focused title or item.
+    this.scrollToTitleOrItem(element);
+  }
+
+  /**
+   * Blur title or item at given index.
+   *
+   * @param {number} titleOrItemIndex - Title or item index.
+   */
+  blurTitleOrItem(titleOrItemIndex) {
+    // check for valid index.
+    if (titleOrItemIndex === null) {
+      return;
+    }
+
+    const { element, type } = this.searchResults[titleOrItemIndex];
+
+    if (!element || !type) {
+      return;
+    }
+
+    // blur title or item.
+    if (type === 'title') {
+      element.classList.remove(SidebarFilter.CSS.sectionTitleSelected);
+    } else if (type === 'item') {
+      element.classList.remove(SidebarFilter.CSS.sectionListItemSlelected);
+    }
+  }
+
+  /**
+   * Scroll to title or item.
+   *
+   * @param {HTMLElement} titleOrItem - Title or item element.
+   */
+  scrollToTitleOrItem(titleOrItem) {
+    // check if it's visible.
+    const rect = titleOrItem.getBoundingClientRect();
+    let elemTop = rect.top;
+    let elemBottom = rect.bottom;
+    const halfOfViewport = window.innerHeight / 2;
+    const scrollTop = this.sidebarContent.scrollTop;
+
+    // scroll top if item is not visible.
+    if (elemTop < HEADER_HEIGHT) {
+      // scroll half viewport up.
+      const nextTop = scrollTop - halfOfViewport;
+
+      // check if element visible after scroll.
+      elemTop = (elemTop + nextTop) < HEADER_HEIGHT ? elemTop : nextTop;
+      this.sidebarContent.scroll({
+        top: elemTop,
+        behavior: 'smooth',
+      });
+    } else if (elemBottom > window.innerHeight) {
+      // scroll bottom if item is not visible.
+      // scroll half viewport down.
+      const nextDown = halfOfViewport + scrollTop;
+
+      // check if element visible after scroll.
+      elemBottom = (elemBottom - nextDown) > window.innerHeight ? elemBottom : nextDown;
+      this.sidebarContent.scroll({
+        top: elemBottom,
+        behavior: 'smooth',
+      });
+    }
+  }
+
+  /**
+   * filter sidebar items.
+   *
+   * @param {HTMLElement} section - Section element.
+   * @param {string} searchValue - Search value.
+   */
+  filterSection(section, searchValue) {
+    // match with section title.
+    const sectionTitle = section.querySelector('.' + SidebarFilter.CSS.sectionTitle);
+    const sectionList = section.querySelector('.' + SidebarFilter.CSS.sectionList);
+
+    // check if section title matches.
+    const isTitleMatch = sectionTitle.innerText.trim().toLowerCase()
+      .indexOf(searchValue.toLowerCase()) !== -1;
+    const matchResults = [];
+    // match with section items.
+    let isSingleItemMatch = false;
+
+    if (sectionList) {
+      const sectionListItems = sectionList.querySelectorAll('.' + SidebarFilter.CSS.sectionListItem);
+
+      sectionListItems.forEach(item => {
+        if (item.innerText.trim().toLowerCase()
+          .indexOf(searchValue.toLowerCase()) !== -1) {
```

Comment:
looks like you can move the comparison login to a separate method and get rid of duplicated code.